src/discussions/common/ActionsDropdown.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useContext, useState } from 'react';
+import PropTypes from 'prop-types';
+
+import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
+import { AppContext } from '@edx/frontend-platform/react';
+import {
+  Button, Dropdown, Icon, IconButton, ModalPopup,
+} from '@edx/paragon';
+import { MoreVert } from '@edx/paragon/icons';
+
+import { EditActions } from '../../data/constants';
+import messages from '../messages';
+import { getActions } from '../utils';
+
+function ActionsDropdown({
+  intl,
+  commentOrPost,
+  disabled,
+  onCommentAction,
+}) {
+  const { authenticatedUser } = useContext(AppContext);
+  const [isOpen, setOpen] = useState(false);
+  const dropdownIconRef = React.useRef(null);
+  const actions = getActions(commentOrPost, authenticatedUser);
```

Comment:
> * Why these actions are not passed from the parent component?
> 
> * Why `ActionsDropdown` is aware & responsible for fetching thread actions?

It's a specialised component designed only for handling actions available for a post/comment, so it needs to be aware of the structure of the data `getActions` returns since it has to render it. 

On the other why should two completely different components need to deal with the structure of actions, which they don't know how to render or deal with? To me it made sense to abstract away all the code related to a comment or posts actions to a separate component which encapsulates all the logic related to actions. 

There is definitely one simplification I can make here, which is to make this a hook. I have made that change.